urth/dashboard/converter/__init__.py

```diff
@@ -304,3 +309,123 @@ def to_thebe_html(path, env_vars, fmt, cwd, template_fn):
     if stderr:
         raise RuntimeError('nbconvert wrote to stderr: {}'.format(stderr))
     return stdout
+
+def get_references(abs_nb_path, version):
+    '''
+    Retrieves the raw references to files, folders, and exclusions
+    from a notebook's markdown cell comments.
+    '''
+    notebook = nbformat.read(abs_nb_path, version)
+    referenced_list = []
+    for cell in notebook.cells:
+        references = _get_references(cell)
+        if references:
+            referenced_list = referenced_list + references
+    return referenced_list
+
+def get_referenced_files(abs_nb_path, version):
+    '''
+    Retrieves the full list of files referenced by a notebook's
+    markdown cell comments, as relative paths. Temporarily changes
+    the current working directory when called.
+    '''
+    expanded = _expand_references(os.path.dirname(abs_nb_path), get_references(abs_nb_path, version))
+    return expanded
+
+def _get_references(cell):
+    '''
+    Retrieves the list of references from a cell, according to
+    an expected way of identifying and separating them.
+    '''
+    referenced = None
+    # invisible after execution: unrendered HTML comment
+    if cell.get('cell_type').startswith('markdown') and cell.get('source').startswith('<!--associate:'):
+        referenced = []
+        lines = cell.get('source')[14:].splitlines()
+        for line in lines:
+            if line.startswith('-->'):
+                break
+            # Trying to go out of the current directory leads to
+            # trouble when deploying
+            if line.find('../') < 0 and not line.startswith('#'):
+                referenced.append(line)
+    # visible after execution: rendered as a code element within a pre element
+    elif cell.get('cell_type').startswith('markdown') and cell.get('source').find('```') >= 0:
```

Comment:
Is there a reason the comment syntax has a prefix but not the triple back tick (fenced code block)? Any concerns about parsing all fenced code blocks and checking for existence of files on disk. For example, what if I have a fenced block I'm using to list a set of file names purely for documentation purposes, not to have them packed up with the notebook?